### ISSUE TITLE
TFLite Android: Model file will not load. startOffset and declaredLength problems

### ISSUE THREAD

**Issue by andrewginns** [tensorflow/tensorflow#21245]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: Pixel 2XL, Android studio emulator
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: 1.9
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: 0.11.0
- **GCC/Compiler version (if compiling from source)**: 5.4
- **CUDA/cuDNN version**: n/a
- **GPU model and memory**: n/a
- **Exact command to reproduce**: Loading TFLite model in an Android application

Posted as an issue here based on comment from https://stackoverflow.com/questions/51341554/tflite-android-model-file-will-not-load-startoffset-and-declaredlength-problem?noredirect=1#comment90097492_51341554

I'm having issues with loading a TFLite model using the MappedByteBuffer method from the Tensorflow-for-poets-2 TFLite tutorial.

```
private MappedByteBuffer loadModelFile(Activity activity,String MODEL_FILE) throws IOException {
    AssetFileDescriptor fileDescriptor = activity.getAssets().openFd(MODEL_FILE);
    FileInputStream inputStream = new FileInputStream(fileDescriptor.getFileDescriptor());
    FileChannel fileChannel = inputStream.getChannel();
    long startOffset = fileDescriptor.getStartOffset();
    long declaredLength = fileDescriptor.getDeclaredLength();
    return fileChannel.map(FileChannel.MapMode.READ_ONLY, startOffset, declaredLength);
```

In particular the model I have converted with the tflite_convert (formerly toco) tools fails when the fileChannel.map is returned. The model is a floating point TFLite mode.

The problem seems to be caused by the startOffset and declaredLength variables. The error I receive in the logcat is
```
7525-7525/dp.thexor A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 7525 (dp.thexor), pid 7525 (dp.thexor)
If I fix these values to ones from a model that successfully loads then the method successfully returns the fileChannel.map. The values from my model are startOffset = 2846788 declaredLength = 45525464
```

I know that my TFlite model can successfully be mapped to memory as the tensorflow/contrib/lite/tools/benchmark:benchmark_model is able to benchmark it.

I'd try to load a quantized TFLite model but currently my model contains ops that do not have quantized equivalents (transpose_conv).

What could be causing this?

My .tflite model can be found here: https://github.com/andrewginns/CycleGAN-Tensorflow-PyTorch/releases/download/tf1.7-py3.6.4/float.tflite

**Comment by andrehentz** [comment]
Hi @andrewginns Have you tried using your model with the [TF Lite demo app](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/lite/java/demo/app)? (Trying to find out if this affect only tensorflow for poets or not)

It looks like your model is OK and can run with benchmark_model, for example:
```
$ bazel run -c opt tensorflow/contrib/lite/tools/benchmark:benchmark_model -- --graph=`realpath ~/Downloads/float.tflite`
```

**Comment by andrewginns** [comment]
Hi @andrehentz yeah I have tried with the TF Lite demo app. I went further and tried with a minimal app that should just load the model file into memory but that throws the same error.

That's what confuses me too. The TF Lite model benchmarks work on both desktop and mobile. I'm also able to view the layers of the model using the html visualiser.

**Comment by andrehentz** [comment]
Maybe you can narrow down the crash to a line of native code somewhere? If so, could you share a link?

**Comment by shashishekhar** [comment]
I tried your model it fails in the model verification:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/java/src/main/native/nativeinterpreterwrapper_jni.cc#L302

Can you share the original model before conversion, I will like to see what is going wrong here.

**Comment by andrewginns** [comment]
@shashishekhar

Thanks for looking into it. Here's my model before conversion. 

https://github.com/andrewginns/CycleGAN-Tensorflow-PyTorch/releases/download/tf1.7-py3.6.4/frozen-graph.pb

**Comment by shashishekhar** [comment]
@andrewginns : Quick update, I was able to reproduce the issue, I am working with @miaout17  to check if this is due to any alignment issues.

**Comment by andrewginns** [comment]
@shashishekhar Much appreciated

**Comment by tensorflowbutler** [comment]
Nagging Assignees @miaout17, @jdduke, @shashishekhar: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by shashishekhar** [comment]
@andrewginns : We did a few alignment related fixes and I reconverted the model, I was able to load it without issues, let me know if you are still facing issues. Use tflite_convert for converting the model.

Please reopen if you are still facing the issue.
